docs/architecture/adr-0012-map-component-api.md

```diff
@@ -0,0 +1,380 @@
+# Map component ADR
+## Context
+The current leaflet map implementation was intentionally made as a very thin wrapper. We've developed enough code around it to quantify what we need leaflet to do in raster foundry, as well as the shortcomings of our current approach.  
+
+The initial approach tries to control the leaflet map exclusively thorugh bindings to callbacks and properties. We've found that this creates a lot of complexity in the components and controllers due to the way that data must flow through components.  
+
+Ideally, components should use one-way data bindings where possible, and call bound functions in order to change data in a parent. The current way we have things set up requires all of the important state to be centralized in a parent controller, and modified through two-way bindings in child components.  
+
+Adding layers which need to be controlled by a child attribute has necessitated creating a separate binding for each type of interaction, and this has resulted in a bloated leaflet map component.
+
+## Current use cases
+
+### Scene Browser
+#### Attributes used:
+```
+footprint="$ctrl.hoveredScene.dataFootprint"
+bypass-fit-bounds="true"
+proposed-bounds="$ctrl.bounds"
+grid="$ctrl.lastGridResult"
+on-view-change="$ctrl.onViewChange(newBounds, zoom)"
+```
+
+#### Requirements
+Display single polygon on map  
+Fit map view to bounds  
+Display a grid of polygons  
+Callback on view change  
+
+### Scene Detail & Project Scene Detail
+#### Attributes used:
+```
+footprint="$ctrl.scene.footprint"
+static="true",
+```
+#### Requirements
+Display single polygon on map  
+Option to disable map interaction  
+Option to fit map to polygon  
```

Comment:
we need the ability to fit the map to it. so maybe change to "Ability to always fit map to a given polygon or set of polygons"